plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/image/block.json

```diff
@@ -5,7 +5,7 @@
 	"category": "woocommerce-product-elements",
 	"attributes": {
 		"showProductLink": { "type": "boolean", "default": true },
-		"showSaleBadge": { "type": "boolean", "default": true },
+		"showSaleBadge": { "type": "boolean", "default": false },
```

Comment:
I think I'd consider removing the attribute completely and use `migrate` function from [Deprecation API](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/) to map the flag to the On Sale Badge inner blog, of `true`.